content/blog/2023-09-07-making-connecting-to-localstack-easier/index.md

```diff
@@ -0,0 +1,173 @@
+---
+title: How we are making connecting to LocalStack easier
+date: 2023-09-07
+lastmod: 2023-09-26
+tags:
+- news
+contributors:
+- Simon Walker
+- Daniel Fangl
+---
+
+Connecting your application code to LocalStack is not always easy.
+In this series we will outline some of the ways we have made connectivity within, and <u><i>to</i></u> LocalStack easier.
+
+<!-- picture -->
+
+1. In this post, we will talk about how we have made connecting to LocalStack easier.
+2. In the next post, we will demonstrate a debugging utility that we have built, to help diagnose network connectivity problems.
+3. In the final post, we will show how we have simplified the configuration of the LocalStack container.
+
+## Connectivity to the LocalStack container
+
+LocalStack normally runs in a container, meaning that it is isolated from the host system.
+By default, LocalStack _publishes_ its edge port (usually 4566) to the host.
+Publishing a port means that a ports on the host forwards network communications to the LocalStack container.
+Requests made to `localhost:4566` are then forwarded to the LocalStack container.
+
+This works well when interacting from the host, for example using `awslocal` commands.
+It does not work when trying to connect to LocalStack from your own containers, or LocalStack compute resources such as Lambda functions or ECS containers.
+In the past, some people have suggested connecting their application containers to the host network (`--network host`) or by making requests to `host.docker.internal:4566`.
+In some cases, using the host networking solves the problem, but it raises other ones:
+
+* If SSL is used, then certificate validation must be turned off since LocalStack does not present a valid certificate for the domain used(either `localhost` or `host.docker.internal`).
```

Comment:
I think this sentence might be conceived that the `(either localhost or host.docker.internal)` are the valid names for the certificate? Also, the `localhost` certificate problem is the same when connecting from the host to the container when using a bridge network, something we might (or not, if it gets too long) want to mention above as well.

Reply:
Also, a space should be there before the parenthesis.